RecoHGCal/TICL/plugins/HGCDoublet.h

```diff
@@ -0,0 +1,177 @@
+// Author: Felice Pantaleo - felice.pantaleo@cern.ch
+// Date: 11/2018
+// Copyright CERN
+
+#ifndef __RecoHGCal_TICL_HGCDoublet_H__
+#define __RecoHGCal_TICL_HGCDoublet_H__
+
+#include <cmath>
+#include <vector>
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "DataFormats/CaloRecHit/interface/CaloCluster.h"
+
+class HGCDoublet {
+ public:
+  using HGCntuplet = std::vector<unsigned int>;
+
+  HGCDoublet(const int innerClusterId, const int outerClusterId, const int doubletId,
+             const std::vector<reco::CaloCluster> *layerClusters)
+      : layerClusters_(layerClusters),
+        theDoubletId_(doubletId),
+        innerClusterId_(innerClusterId),
+        outerClusterId_(outerClusterId),
+        innerR_((*layerClusters)[innerClusterId].position().r()),
+        outerR_((*layerClusters)[outerClusterId].position().r()),
+        innerX_((*layerClusters)[innerClusterId].x()),
+        outerX_((*layerClusters)[outerClusterId].x()),
+        innerY_((*layerClusters)[innerClusterId].y()),
+        outerY_((*layerClusters)[outerClusterId].y()),
+        innerZ_((*layerClusters)[innerClusterId].z()),
+        outerZ_((*layerClusters)[outerClusterId].z()),
+        alreadyVisited_(false) {}
+
+  double getInnerX() const { return innerX_; }
+
+  double getOuterX() const { return outerX_; }
+
+  double getInnerY() const { return innerY_; }
+
+  double getOuterY() const { return outerY_; }
+
+  double getInnerZ() const { return innerZ_; }
+
+  double getOuterZ() const { return outerZ_; }
+
+  double getInnerR() const { return innerR_; }
+
+  double getOuterR() const { return outerZ_; }
+
+  int getInnerClusterId() const { return innerClusterId_; }
+
+  int getOuterClusterId() const { return outerClusterId_; }
+
+  void tagAsOuterNeighbor(unsigned int otherDoublet) { outerNeighbors_.push_back(otherDoublet); }
+
+  void tagAsInnerNeighbor(unsigned int otherDoublet) { innerNeighbors_.push_back(otherDoublet); }
+
+  bool checkCompatibilityAndTag(std::vector<HGCDoublet> &allDoublets,
+                                const std::vector<int> &innerDoublets, float minCosTheta,
+                                float minCosPointing = 1., bool debug = false) {
+    int nDoublets = innerDoublets.size();
+    int constexpr VSIZE = 4;
```

Comment:
are there good reasons to keep this and other complex code in this file inlined?